### DISCUSSION TITLE
Pest local build

### DISCUSSION THREAD

**Discussion by paopaowowwow** [pest-parser/pest#938] [General]
I am new to pest and rust, I tried to use "cargo add pest" to install pest for my rust project, however, it doesn't work, error as below, so can I download the source code and proceed with local build? if yes, how? thank you
warning: spurious network error (3 tries remaining): [6] Couldn't resolve host name (Could not resolve host: index.crates.io)
warning: spurious network error (2 tries remaining): [6] Couldn't resolve host name (Could not resolve host: index.crates.io)
warning: spurious network error (1 tries remaining): [6] Couldn't resolve host name (Could not resolve host: index.crates.io)
error: failed to query replaced source registry `crates-io`

Caused by:
  download of config.json failed

Caused by:
  failed to download from `https://index.crates.io/config.json`

Caused by:
  [6] Couldn't resolve host name (Could not resolve host: index.crates.io)

**Comment by Jamalam360** [comment]
This is just a network error, either with your connection or there was an issue with crates.io.

It should work if you try again and ensure your connection is ok

**Reply by paopaowowwow** [reply]
thank you for your reply, do you know how to fix the connection? I downloaded pest source codes and try to proceed with cargo build, then the error occurs. but I can browse crates.io. with chrome. when I try to run cargo install <library name> in terminal, then above error happens.   not sure if our company blocked the site.

**Reply by Jamalam360** [reply]
It might be blocked but it'd be weird to have the index blocked and not crates itself. Can you go to https://index.crates.io/config.json?

**Reply by paopaowowwow** [reply]
I can view it with browse.   and I can view https://crates.io/ as well.  proceed with cargo build in project, then errors happen, do I need to update anything in .cargo/confige.toml?
{
    "dl": "https://crates.io/api/v1/crates",
    "api": "https://crates.io"
}
